main/parse.c

```diff
@@ -3651,6 +3636,23 @@ extern bool parseFileWithMio (const char *const fileName, MIO *mio)
 	return tagFileResized;
 }
 
+extern void *parseRawBuffer(const char *fileName, unsigned char *buffer,
+			 size_t bufferSize, const langType language)
+{
+	MIO *mio = NULL;
+	void *rval = NULL;
+
+	if (buffer)
+		mio = mio_new_memory (buffer, bufferSize, NULL, NULL);
+	parseMio (fileName, language, mio, false, &rval);
+	mio_free (mio);
+
+	if (buffer)
+		mio_free (mio);
+
+	return rval;
+}
+
```

Comment:
We could remove parseRawBuffer() from ctags completely and just export parseMio() which the client app would use directly.

Reply:
It means we have to export mio API from ctags.
In the future, it is a good idea. However, in short term, it is not a good idea because, as I wrote, I introduced changes to mio ctags locally. So I think you cannot pass a mio instance in Geany to ctags.
As a temporary solution, we have to do
(1) rename mio in ctags to cmio or something, and
(2) export parseRawBuffer as non-mio API.

Here, I would like to introduce parseRawBuffer as the first step.
Is this acceptable?

Reply:
Yeah, totally. I just suggested it as an option and maybe exposing MIO really isn't a good idea at all.

Reply:
I'm not sure exposing MIO makes much sense, it's only useful if the caller would actually do something we don't with MIO.  That could include using MIO to create the buffer to be parsed (e.g. in memory), or creating it with a constructor we don't expose (e.g. `mio_new_fp()` in combination with [`fdopen()`](http://man7.org/linux/man-pages/man3/fdopen.3p.html)), but that's about it.